### PR TITLE
update docker to multi-stage build

### PR DESCRIPTION
**Pull request by joshgoebel**
- reduces size from ~900mb to ~90mb

### CODE REVIEW

File: dev.Dockerfile [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
-           EXERCISM_DOCKER=true EXERCISM_ENV=development bundle exec setup_exercism_config && \
-           APP_ENV=development bundle exec rackup -p 3021 --host 0.0.0.0
+
+RUN gem install bundler:2.1.4
+COPY --from=gembuilder /usr/local/bundle/ /usr/local/bundle/
+
+# copy the source as late as possible to maximize cache
+COPY . .
+
+ENV CONTAINER_NAME=tooling-orchestrator
```
Comment (iHiD) [inline dev.Dockerfile:27]: What is this used for?

File: dev.Dockerfile [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+#############
+## Stage 1 ##
+#############
+FROM ruby:2.6.6-alpine3.12 as gembuilder
 
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
-    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
-    curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
-    apt-get install -y cmake
+RUN apk add --no-cache --update build-base cmake
```
Comment (iHiD) [inline dev.Dockerfile:6]: Why `--no-cache`?

Comment (joshgoebel) [reply dev.Dockerfile:6]: I guess now that it's stage 1 it matters less, but it still isn't hurting anything really (just using less space).

File: dev.Dockerfile (hunk 1 above)
Comment (joshgoebel) [reply dev.Dockerfile:27]: Shell prompt, the container knowing "what am I", etc.

File: dev.Dockerfile (hunk 2 above)
Comment (iHiD) [reply dev.Dockerfile:6]: Ah, so that says "don't cache things" rather than "don't use the cache"?

File: dev.Dockerfile (hunk 1 above)
Comment (iHiD) [reply dev.Dockerfile:27]: Is this an official docker thing or something you've made up?

Comment (joshgoebel) [reply dev.Dockerfile:27]: Something I made up.

File: dev.Dockerfile (hunk 2 above)
Comment (joshgoebel) [reply dev.Dockerfile:6]: AFAIK yes it's space saving.

File: dev.Dockerfile (hunk 1 above)
Comment (iHiD) [reply dev.Dockerfile:27]: So, this is here so that within the shell, you can do `echo $CONTAINER_NAME`?

Comment (joshgoebel) [reply dev.Dockerfile:27]: Yes or use it for any other purpose.  It seems like something that could be useful for container diag, etc.

**Comment by iHiD** [comment]
Also amazing work, thanks @joshgoebel 🎉

**Pull request merged**